tests/app/dao/test_templates_dao.py

```diff
@@ -957,3 +959,67 @@ def test_template_with_provider_id_persists_provider_id(
     assert notify_db_session.session.get(Template, template.id).provider_id == provider.id
     # Teardown
     template_cleanup(notify_db_session.session, template.id)
+
+
+class TestDAOGetTemplateById:
+    def test_dao_get_template_by_id_with_no_version(
+        self,
+        sample_template: Callable[..., Any],
+    ):
+        template = sample_template()
+
+        result = dao_get_template_by_id(template.id)
+
+        assert result == template
+        assert isinstance(result, Template)
+        assert result.id == template.id
+        assert result.version == 1
+
+    def test_dao_get_template_by_id_with_specific_version(
+        self,
+        sample_template: Callable[..., Any],
+    ):
+        template = sample_template()
+
+        template.content = 'Updated content'
+        dao_update_template(template)
```

Comment:
This application code is not intended to be under test.  A better test implementation would be to add and commit the change using notify_db_session.